src/nonePublicStuff.ts

```diff
@@ -0,0 +1,345 @@
+  }
+
+  // maybe pass here original and not intermediate values?
+  let responseIntermediate = {
+    response: rawResponse,
+    type: requestIntermediate.type,
+    resource: requestIntermediate.resource,
+    params: requestIntermediate.params
+  };
+
+  if (configuration && configuration.responseMutator) {
+    responseIntermediate = configuration.responseMutator(
+      responseIntermediate,
+      requestIntermediate
+    );
+  }
+
+  const oneMoreIntermediateResponse = convertHTTPResponse(
+    responseIntermediate.response,
+    requestIntermediate
+  );
+
+  // in some conditions, we want to fetch fresh copy of the data from nest/crud.
+  switch (requestIntermediate.type) {
+    case "CREATE":
+    case "UPDATE":
+      return makeNestjsxCrudRequest(
+        apiUrl,
+        httpClient,
+        configuration,
+        "GET_ONE",
+        resource,
+        { id: oneMoreIntermediateResponse.data.id }
+      );
+
+    case "UPDATE_MANY":
+      return Promise.all(
+        oneMoreIntermediateResponse.data.map(({ id }: { id: any }) =>
+          makeNestjsxCrudRequest(
+            apiUrl,
+            httpClient,
+            configuration,
+            "GET_ONE",
+            resource,
+            { id }
+          )
+        )
+      ).then((all: any) => ({
+        data: all
+      }));
+
+    default:
+      return oneMoreIntermediateResponse;
+  }
+}
+
+/** @internal */
+export function convertHTTPResponse(
+  rawResponse: any,
+  request: ClientRequestType
+) {
+  let toReturn: any = Array.isArray(rawResponse)
+    ? { data: rawResponse.map(entry => entry.json) }
+    : { data: rawResponse.json };
+
+  switch (request.type) {
+    case DELETE_MANY:
+      toReturn = {
+        data: request.params.ids
+      };
+      break;
+
+    case GET_LIST:
+    case GET_MANY_REFERENCE:
+      toReturn = {
+        data: rawResponse.json.data,
+        total: rawResponse.json.total
+      };
+  }
+
+  return toReturn;
+}
+
+/** @internal */
+export function composeFilter(paramsFilter: any) {
+  const flatFilter = fetchUtils.flattenObject(paramsFilter);
+  const filter = Object.keys(flatFilter).map(key => {
+    const splitKey = key.split("||");
+    const ops = splitKey[1] ? splitKey[1] : CondOperator.CONTAINS;
+    let field = splitKey[0];
+
+    if (field.indexOf("_") === 0 && field.indexOf(".") > -1) {
+      field = field.split(/\.(.+)/)[1];
+    }
+    return { field, operator: ops, value: flatFilter[key] };
+  });
+  return filter;
+}
+
+/** @internal */
+export const MAGIC_SEPARATOR = "_._._._";
+
+/** @internal */
+export function extractRealResourceAndParams(resourceMaybeWithEncoded: string) {
+  if (!resourceMaybeWithEncoded.includes(MAGIC_SEPARATOR)) {
+    return {
+      realResource: resourceMaybeWithEncoded
+    };
+  }
+
+  const [realResource, paramsStr] = resourceMaybeWithEncoded.split(
+    MAGIC_SEPARATOR
+  );
+
+  try {
+    const integratedParams: Pick<
+      import("@nestjsx/crud-request").CreateQueryParams,
```

Comment:
import declarations may only appear at the top level of a module

Reply:
this is type import, not module import